{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style type=\"text/css\">\n",
       ".container {\n",
       "    width: 100%;\n",
       "}\n",
       "</style>\n"
      ],
      "text/plain": [
       "<IPython.core.display.HTML object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%run setup.ipynb"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# IR alleles - supplementary table\n",
    "per sample allele counts for known IR loci - kdr, ace1, gste2 \n",
    "- GSTE and ace1 variants are only present in the raw data - need to use this and provide caveats\n",
    "- this could be merged with the main meta data table?\n",
    "- are there any more loci we could include?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ox_code</th>\n",
       "      <th>src_code</th>\n",
       "      <th>population</th>\n",
       "      <th>country</th>\n",
       "      <th>location</th>\n",
       "      <th>site</th>\n",
       "      <th>contributor</th>\n",
       "      <th>contact</th>\n",
       "      <th>year</th>\n",
       "      <th>m_s</th>\n",
       "      <th>sex</th>\n",
       "      <th>n_sequences</th>\n",
       "      <th>mean_coverage</th>\n",
       "      <th>ebi_sample_acc</th>\n",
       "      <th>latitude</th>\n",
       "      <th>longitude</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AA0040-C</td>\n",
       "      <td>Twifo_Praso__E2</td>\n",
       "      <td>GHcol</td>\n",
       "      <td>Ghana</td>\n",
       "      <td>Twifo Praso</td>\n",
       "      <td>Twifo Praso</td>\n",
       "      <td>David Weetman</td>\n",
       "      <td>David Weetman</td>\n",
       "      <td>2012</td>\n",
       "      <td>M</td>\n",
       "      <td>F</td>\n",
       "      <td>95033368</td>\n",
       "      <td>30.99</td>\n",
       "      <td>ERS311878</td>\n",
       "      <td>5.60858</td>\n",
       "      <td>-1.54926</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AA0041-C</td>\n",
       "      <td>Twifo_Praso__H3</td>\n",
       "      <td>GHcol</td>\n",
       "      <td>Ghana</td>\n",
       "      <td>Twifo Praso</td>\n",
       "      <td>Twifo Praso</td>\n",
       "      <td>David Weetman</td>\n",
       "      <td>David Weetman</td>\n",
       "      <td>2012</td>\n",
       "      <td>M</td>\n",
       "      <td>F</td>\n",
       "      <td>95843804</td>\n",
       "      <td>31.70</td>\n",
       "      <td>ERS311886</td>\n",
       "      <td>5.60858</td>\n",
       "      <td>-1.54926</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AA0051-C</td>\n",
       "      <td>Twifo_Praso__C1</td>\n",
       "      <td>GHcol</td>\n",
       "      <td>Ghana</td>\n",
       "      <td>Twifo Praso</td>\n",
       "      <td>Twifo Praso</td>\n",
       "      <td>David Weetman</td>\n",
       "      <td>David Weetman</td>\n",
       "      <td>2012</td>\n",
       "      <td>M</td>\n",
       "      <td>F</td>\n",
       "      <td>99910300</td>\n",
       "      <td>33.33</td>\n",
       "      <td>ERS311871</td>\n",
       "      <td>5.60858</td>\n",
       "      <td>-1.54926</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AA0052-C</td>\n",
       "      <td>Twifo_Praso__F2</td>\n",
       "      <td>GHcol</td>\n",
       "      <td>Ghana</td>\n",
       "      <td>Twifo Praso</td>\n",
       "      <td>Twifo Praso</td>\n",
       "      <td>David Weetman</td>\n",
       "      <td>David Weetman</td>\n",
       "      <td>2012</td>\n",
       "      <td>M</td>\n",
       "      <td>F</td>\n",
       "      <td>100899280</td>\n",
       "      <td>32.29</td>\n",
       "      <td>ERS311879</td>\n",
       "      <td>5.60858</td>\n",
       "      <td>-1.54926</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AA0053-C</td>\n",
       "      <td>Twifo_Praso__A4</td>\n",
       "      <td>GHcol</td>\n",
       "      <td>Ghana</td>\n",
       "      <td>Twifo Praso</td>\n",
       "      <td>Twifo Praso</td>\n",
       "      <td>David Weetman</td>\n",
       "      <td>David Weetman</td>\n",
       "      <td>2012</td>\n",
       "      <td>M</td>\n",
       "      <td>F</td>\n",
       "      <td>104791234</td>\n",
       "      <td>34.14</td>\n",
       "      <td>ERS311887</td>\n",
       "      <td>5.60858</td>\n",
       "      <td>-1.54926</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    ox_code         src_code population country     location         site  \\\n",
       "0  AA0040-C  Twifo_Praso__E2      GHcol   Ghana  Twifo Praso  Twifo Praso   \n",
       "1  AA0041-C  Twifo_Praso__H3      GHcol   Ghana  Twifo Praso  Twifo Praso   \n",
       "2  AA0051-C  Twifo_Praso__C1      GHcol   Ghana  Twifo Praso  Twifo Praso   \n",
       "3  AA0052-C  Twifo_Praso__F2      GHcol   Ghana  Twifo Praso  Twifo Praso   \n",
       "4  AA0053-C  Twifo_Praso__A4      GHcol   Ghana  Twifo Praso  Twifo Praso   \n",
       "\n",
       "     contributor        contact  year m_s sex  n_sequences  mean_coverage  \\\n",
       "0  David Weetman  David Weetman  2012   M   F     95033368          30.99   \n",
       "1  David Weetman  David Weetman  2012   M   F     95843804          31.70   \n",
       "2  David Weetman  David Weetman  2012   M   F     99910300          33.33   \n",
       "3  David Weetman  David Weetman  2012   M   F    100899280          32.29   \n",
       "4  David Weetman  David Weetman  2012   M   F    104791234          34.14   \n",
       "\n",
       "  ebi_sample_acc  latitude  longitude  \n",
       "0      ERS311878   5.60858   -1.54926  \n",
       "1      ERS311886   5.60858   -1.54926  \n",
       "2      ERS311871   5.60858   -1.54926  \n",
       "3      ERS311879   5.60858   -1.54926  \n",
       "4      ERS311887   5.60858   -1.54926  "
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_meta = pd.read_csv('../meta/ag1000g/phase2_samples.meta.txt', sep='\\t')\n",
    "df_meta.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "callset_fn = '../phase2.AR1/variation/main/zarr/all/ag1000g.phase2.ar1/'\n",
    "callset = zarr.open_group(callset_fn) "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## kdr - 995F/S/Y"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "genk = allel.GenotypeDaskArray(callset['2L']['calldata/genotype'])\n",
    "posk = allel.SortedIndex(callset['2L']['variants']['POS'][:])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "#kdr - 995F and S\n",
    "F = genk[posk.locate_range(2422652, 2422652)]\n",
    "Y = genk[posk.locate_range(2429745, 2429745)]\n",
    "S = genk[posk.locate_range(2422651, 2422651)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "kf = []\n",
    "ky = []\n",
    "ks = []\n",
    "for f, y, s in zip(np.asarray(F)[0], np.asarray(Y)[0], np.asarray(S)[0]):\n",
    "    kf.append((f[0] + f[1]))\n",
    "    ky.append((y[0] + y[1]))\n",
    "    ks.append((s[0] + s[1]))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## ace-1 - G119S"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "gena = allel.GenotypeDaskArray(callset['2R']['calldata/GT'])\n",
    "posa = allel.SortedIndex(callset['2R']['variants']['POS'][:])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "A = gena[posa.locate_range(3492074, 3492074)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "a1 = []\n",
    "for a in np.asarray(A)[0]:\n",
    "    a1.append((a[0] + a[1]))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## GSTE2 - I114T"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "geng = allel.GenotypeDaskArray(callset['3R']['calldata/genotype'])\n",
    "posg = allel.SortedIndex(callset['3R']['variants']['POS'][:])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "G = geng[posg.locate_range(28598166, 28598166)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "gi = []\n",
    "for g in np.asarray(G)[0]:\n",
    "    gi.append((g[0] + g[1]))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Supplementary table - per sample allele count - separate table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "S      654\n",
       "M      283\n",
       "M/S      1\n",
       "Name: m_s, dtype: int64"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sp = df_meta['m_s']\n",
    "sp.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "gambiae     654\n",
       "coluzzii    283\n",
       "hybrid        1\n",
       "Name: m_s, dtype: int64"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sp.replace(['S', 'M', 'M/S'], ['gambiae', 'coluzzii', 'hybrid'], inplace=True)\n",
    "sp.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ox_code</th>\n",
       "      <th>population</th>\n",
       "      <th>country</th>\n",
       "      <th>location</th>\n",
       "      <th>site</th>\n",
       "      <th>year</th>\n",
       "      <th>species</th>\n",
       "      <th>latitude</th>\n",
       "      <th>longitude</th>\n",
       "      <th>vgsc-L995F</th>\n",
       "      <th>vgsc-N1570Y</th>\n",
       "      <th>vgsc-L995S</th>\n",
       "      <th>ace1-G119S</th>\n",
       "      <th>gste2-I114T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AA0040-C</td>\n",
       "      <td>GHcol</td>\n",
       "      <td>Ghana</td>\n",
       "      <td>Twifo Praso</td>\n",
       "      <td>Twifo Praso</td>\n",
       "      <td>2012</td>\n",
       "      <td>coluzzii</td>\n",
       "      <td>5.60858</td>\n",
       "      <td>-1.54926</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AA0041-C</td>\n",
       "      <td>GHcol</td>\n",
       "      <td>Ghana</td>\n",
       "      <td>Twifo Praso</td>\n",
       "      <td>Twifo Praso</td>\n",
       "      <td>2012</td>\n",
       "      <td>coluzzii</td>\n",
       "      <td>5.60858</td>\n",
       "      <td>-1.54926</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AA0051-C</td>\n",
       "      <td>GHcol</td>\n",
       "      <td>Ghana</td>\n",
       "      <td>Twifo Praso</td>\n",
       "      <td>Twifo Praso</td>\n",
       "      <td>2012</td>\n",
       "      <td>coluzzii</td>\n",
       "      <td>5.60858</td>\n",
       "      <td>-1.54926</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AA0052-C</td>\n",
       "      <td>GHcol</td>\n",
       "      <td>Ghana</td>\n",
       "      <td>Twifo Praso</td>\n",
       "      <td>Twifo Praso</td>\n",
       "      <td>2012</td>\n",
       "      <td>coluzzii</td>\n",
       "      <td>5.60858</td>\n",
       "      <td>-1.54926</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AA0053-C</td>\n",
       "      <td>GHcol</td>\n",
       "      <td>Ghana</td>\n",
       "      <td>Twifo Praso</td>\n",
       "      <td>Twifo Praso</td>\n",
       "      <td>2012</td>\n",
       "      <td>coluzzii</td>\n",
       "      <td>5.60858</td>\n",
       "      <td>-1.54926</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    ox_code population country     location         site  year   species  \\\n",
       "0  AA0040-C      GHcol   Ghana  Twifo Praso  Twifo Praso  2012  coluzzii   \n",
       "1  AA0041-C      GHcol   Ghana  Twifo Praso  Twifo Praso  2012  coluzzii   \n",
       "2  AA0051-C      GHcol   Ghana  Twifo Praso  Twifo Praso  2012  coluzzii   \n",
       "3  AA0052-C      GHcol   Ghana  Twifo Praso  Twifo Praso  2012  coluzzii   \n",
       "4  AA0053-C      GHcol   Ghana  Twifo Praso  Twifo Praso  2012  coluzzii   \n",
       "\n",
       "   latitude  longitude  vgsc-L995F  vgsc-N1570Y  vgsc-L995S  ace1-G119S  \\\n",
       "0   5.60858   -1.54926           2            0           0           0   \n",
       "1   5.60858   -1.54926           2            0           0           0   \n",
       "2   5.60858   -1.54926           2            0           0           0   \n",
       "3   5.60858   -1.54926           1            0           0           0   \n",
       "4   5.60858   -1.54926           1            0           0           0   \n",
       "\n",
       "   gste2-I114T  \n",
       "0            0  \n",
       "1            2  \n",
       "2            0  \n",
       "3            1  \n",
       "4            0  "
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sample_df = pandas.DataFrame()\n",
    "sample_df['ox_code'] = df_meta['ox_code']\n",
    "sample_df['population'] = df_meta['population']\n",
    "sample_df['country'] = df_meta['country']\n",
    "sample_df['location'] = df_meta['location']\n",
    "sample_df['site'] = df_meta['site']\n",
    "sample_df['year'] = df_meta['year']\n",
    "sample_df['species'] = sp\n",
    "sample_df['latitude'] = df_meta['latitude']\n",
    "sample_df['longitude'] = df_meta['longitude']\n",
    "sample_df['vgsc-L995F'] = kf\n",
    "sample_df['vgsc-N1570Y'] = ky\n",
    "sample_df['vgsc-L995S'] = ks\n",
    "sample_df['ace1-G119S'] = a1\n",
    "sample_df['gste2-I114T'] = gi\n",
    "sample_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "sample_df.to_csv('../tables/sample_IR_counts.csv', sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Supplementary table - per sample allele count - combined with meta table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ox_code</th>\n",
       "      <th>vgsc-L995F</th>\n",
       "      <th>vgsc-N1570Y</th>\n",
       "      <th>vgsc-L995S</th>\n",
       "      <th>ace1-G119S</th>\n",
       "      <th>gste2-I114T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AA0040-C</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AA0041-C</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AA0051-C</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AA0052-C</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AA0053-C</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    ox_code  vgsc-L995F  vgsc-N1570Y  vgsc-L995S  ace1-G119S  gste2-I114T\n",
       "0  AA0040-C           2            0           0           0            0\n",
       "1  AA0041-C           2            0           0           0            2\n",
       "2  AA0051-C           2            0           0           0            0\n",
       "3  AA0052-C           1            0           0           0            1\n",
       "4  AA0053-C           1            0           0           0            0"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "to_join_df = sample_df[['ox_code', 'vgsc-L995F', 'vgsc-N1570Y', 'vgsc-L995S', 'ace1-G119S', 'gste2-I114T']]\n",
    "to_join_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "sample_combined_df = df_meta.merge(to_join_df, on='ox_code')\n",
    "sample_combined_df.to_csv('../tables/sample_IR_counts_meta_combined.csv', sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ox_code</th>\n",
       "      <th>src_code</th>\n",
       "      <th>population</th>\n",
       "      <th>country</th>\n",
       "      <th>location</th>\n",
       "      <th>site</th>\n",
       "      <th>contributor</th>\n",
       "      <th>contact</th>\n",
       "      <th>year</th>\n",
       "      <th>m_s</th>\n",
       "      <th>...</th>\n",
       "      <th>n_sequences</th>\n",
       "      <th>mean_coverage</th>\n",
       "      <th>ebi_sample_acc</th>\n",
       "      <th>latitude</th>\n",
       "      <th>longitude</th>\n",
       "      <th>vgsc-L995F</th>\n",
       "      <th>vgsc-N1570Y</th>\n",
       "      <th>vgsc-L995S</th>\n",
       "      <th>ace1-G119S</th>\n",
       "      <th>gste2-I114T</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AA0040-C</td>\n",
       "      <td>Twifo_Praso__E2</td>\n",
       "      <td>GHcol</td>\n",
       "      <td>Ghana</td>\n",
       "      <td>Twifo Praso</td>\n",
       "      <td>Twifo Praso</td>\n",
       "      <td>David Weetman</td>\n",
       "      <td>David Weetman</td>\n",
       "      <td>2012</td>\n",
       "      <td>coluzzii</td>\n",
       "      <td>...</td>\n",
       "      <td>95033368</td>\n",
       "      <td>30.99</td>\n",
       "      <td>ERS311878</td>\n",
       "      <td>5.60858</td>\n",
       "      <td>-1.54926</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AA0041-C</td>\n",
       "      <td>Twifo_Praso__H3</td>\n",
       "      <td>GHcol</td>\n",
       "      <td>Ghana</td>\n",
       "      <td>Twifo Praso</td>\n",
       "      <td>Twifo Praso</td>\n",
       "      <td>David Weetman</td>\n",
       "      <td>David Weetman</td>\n",
       "      <td>2012</td>\n",
       "      <td>coluzzii</td>\n",
       "      <td>...</td>\n",
       "      <td>95843804</td>\n",
       "      <td>31.70</td>\n",
       "      <td>ERS311886</td>\n",
       "      <td>5.60858</td>\n",
       "      <td>-1.54926</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AA0051-C</td>\n",
       "      <td>Twifo_Praso__C1</td>\n",
       "      <td>GHcol</td>\n",
       "      <td>Ghana</td>\n",
       "      <td>Twifo Praso</td>\n",
       "      <td>Twifo Praso</td>\n",
       "      <td>David Weetman</td>\n",
       "      <td>David Weetman</td>\n",
       "      <td>2012</td>\n",
       "      <td>coluzzii</td>\n",
       "      <td>...</td>\n",
       "      <td>99910300</td>\n",
       "      <td>33.33</td>\n",
       "      <td>ERS311871</td>\n",
       "      <td>5.60858</td>\n",
       "      <td>-1.54926</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AA0052-C</td>\n",
       "      <td>Twifo_Praso__F2</td>\n",
       "      <td>GHcol</td>\n",
       "      <td>Ghana</td>\n",
       "      <td>Twifo Praso</td>\n",
       "      <td>Twifo Praso</td>\n",
       "      <td>David Weetman</td>\n",
       "      <td>David Weetman</td>\n",
       "      <td>2012</td>\n",
       "      <td>coluzzii</td>\n",
       "      <td>...</td>\n",
       "      <td>100899280</td>\n",
       "      <td>32.29</td>\n",
       "      <td>ERS311879</td>\n",
       "      <td>5.60858</td>\n",
       "      <td>-1.54926</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AA0053-C</td>\n",
       "      <td>Twifo_Praso__A4</td>\n",
       "      <td>GHcol</td>\n",
       "      <td>Ghana</td>\n",
       "      <td>Twifo Praso</td>\n",
       "      <td>Twifo Praso</td>\n",
       "      <td>David Weetman</td>\n",
       "      <td>David Weetman</td>\n",
       "      <td>2012</td>\n",
       "      <td>coluzzii</td>\n",
       "      <td>...</td>\n",
       "      <td>104791234</td>\n",
       "      <td>34.14</td>\n",
       "      <td>ERS311887</td>\n",
       "      <td>5.60858</td>\n",
       "      <td>-1.54926</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    ox_code         src_code population country     location         site  \\\n",
       "0  AA0040-C  Twifo_Praso__E2      GHcol   Ghana  Twifo Praso  Twifo Praso   \n",
       "1  AA0041-C  Twifo_Praso__H3      GHcol   Ghana  Twifo Praso  Twifo Praso   \n",
       "2  AA0051-C  Twifo_Praso__C1      GHcol   Ghana  Twifo Praso  Twifo Praso   \n",
       "3  AA0052-C  Twifo_Praso__F2      GHcol   Ghana  Twifo Praso  Twifo Praso   \n",
       "4  AA0053-C  Twifo_Praso__A4      GHcol   Ghana  Twifo Praso  Twifo Praso   \n",
       "\n",
       "     contributor        contact  year       m_s     ...      n_sequences  \\\n",
       "0  David Weetman  David Weetman  2012  coluzzii     ...         95033368   \n",
       "1  David Weetman  David Weetman  2012  coluzzii     ...         95843804   \n",
       "2  David Weetman  David Weetman  2012  coluzzii     ...         99910300   \n",
       "3  David Weetman  David Weetman  2012  coluzzii     ...        100899280   \n",
       "4  David Weetman  David Weetman  2012  coluzzii     ...        104791234   \n",
       "\n",
       "   mean_coverage  ebi_sample_acc latitude  longitude  vgsc-L995F  vgsc-N1570Y  \\\n",
       "0          30.99       ERS311878  5.60858   -1.54926           2            0   \n",
       "1          31.70       ERS311886  5.60858   -1.54926           2            0   \n",
       "2          33.33       ERS311871  5.60858   -1.54926           2            0   \n",
       "3          32.29       ERS311879  5.60858   -1.54926           1            0   \n",
       "4          34.14       ERS311887  5.60858   -1.54926           1            0   \n",
       "\n",
       "   vgsc-L995S  ace1-G119S  gste2-I114T  \n",
       "0           0           0            0  \n",
       "1           0           0            2  \n",
       "2           0           0            0  \n",
       "3           0           0            1  \n",
       "4           0           0            0  \n",
       "\n",
       "[5 rows x 21 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sample_combined_df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## per population allele frequencies"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>vgsc-L995F</th>\n",
       "      <th>vgsc-N1570Y</th>\n",
       "      <th>vgsc-L995S</th>\n",
       "      <th>ace1-G119S</th>\n",
       "      <th>gste2-I114T</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>population</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>AOcol</th>\n",
       "      <td>131</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BFcol</th>\n",
       "      <td>145</td>\n",
       "      <td>31</td>\n",
       "      <td>0</td>\n",
       "      <td>14</td>\n",
       "      <td>38</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BFgam</th>\n",
       "      <td>167</td>\n",
       "      <td>47</td>\n",
       "      <td>0</td>\n",
       "      <td>18</td>\n",
       "      <td>51</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CIcol</th>\n",
       "      <td>130</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>62</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CMgam</th>\n",
       "      <td>312</td>\n",
       "      <td>34</td>\n",
       "      <td>93</td>\n",
       "      <td>0</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            vgsc-L995F  vgsc-N1570Y  vgsc-L995S  ace1-G119S  gste2-I114T\n",
       "population                                                              \n",
       "AOcol              131            0           0           0            1\n",
       "BFcol              145           31           0          14           38\n",
       "BFgam              167           47           0          18           51\n",
       "CIcol              130            0           0          62           20\n",
       "CMgam              312           34          93           0            5"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pop_df = sample_df[['population', 'vgsc-L995F', 'vgsc-N1570Y', 'vgsc-L995S', 'ace1-G119S', 'gste2-I114T']].groupby(['population']).sum()\n",
    "pop_df.reset_index(inplace=True)\n",
    "pop_df.set_index('population', inplace=True)\n",
    "pop_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>vgsc-L995F</th>\n",
       "      <th>vgsc-N1570Y</th>\n",
       "      <th>vgsc-L995S</th>\n",
       "      <th>ace1-G119S</th>\n",
       "      <th>gste2-I114T</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>population</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>AOcol</th>\n",
       "      <td>83.97</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.64</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BFcol</th>\n",
       "      <td>96.67</td>\n",
       "      <td>20.67</td>\n",
       "      <td>0.00</td>\n",
       "      <td>9.33</td>\n",
       "      <td>25.33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>BFgam</th>\n",
       "      <td>90.76</td>\n",
       "      <td>25.54</td>\n",
       "      <td>0.00</td>\n",
       "      <td>9.78</td>\n",
       "      <td>27.72</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CIcol</th>\n",
       "      <td>91.55</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>43.66</td>\n",
       "      <td>14.08</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CMgam</th>\n",
       "      <td>52.53</td>\n",
       "      <td>5.72</td>\n",
       "      <td>15.66</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.84</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>FRgam</th>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GAgam</th>\n",
       "      <td>32.61</td>\n",
       "      <td>0.00</td>\n",
       "      <td>67.39</td>\n",
       "      <td>0.00</td>\n",
       "      <td>28.99</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GHcol</th>\n",
       "      <td>88.18</td>\n",
       "      <td>2.73</td>\n",
       "      <td>0.00</td>\n",
       "      <td>18.18</td>\n",
       "      <td>34.55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GHgam</th>\n",
       "      <td>70.83</td>\n",
       "      <td>4.17</td>\n",
       "      <td>0.00</td>\n",
       "      <td>12.50</td>\n",
       "      <td>33.33</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GM</th>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GNcol</th>\n",
       "      <td>87.50</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>12.50</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GNgam</th>\n",
       "      <td>100.00</td>\n",
       "      <td>8.75</td>\n",
       "      <td>0.00</td>\n",
       "      <td>11.25</td>\n",
       "      <td>15.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GQgam</th>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GW</th>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>KE</th>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>76.04</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>UGgam</th>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>100.00</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            vgsc-L995F  vgsc-N1570Y  vgsc-L995S  ace1-G119S  gste2-I114T\n",
       "population                                                              \n",
       "AOcol            83.97         0.00        0.00        0.00         0.64\n",
       "BFcol            96.67        20.67        0.00        9.33        25.33\n",
       "BFgam            90.76        25.54        0.00        9.78        27.72\n",
       "CIcol            91.55         0.00        0.00       43.66        14.08\n",
       "CMgam            52.53         5.72       15.66        0.00         0.84\n",
       "FRgam             0.00         0.00        0.00        0.00         0.00\n",
       "GAgam            32.61         0.00       67.39        0.00        28.99\n",
       "GHcol            88.18         2.73        0.00       18.18        34.55\n",
       "GHgam            70.83         4.17        0.00       12.50        33.33\n",
       "GM                0.00         0.00        0.00        0.00         0.00\n",
       "GNcol            87.50         0.00        0.00        0.00        12.50\n",
       "GNgam           100.00         8.75        0.00       11.25        15.00\n",
       "GQgam             0.00         0.00        0.00        0.00         0.00\n",
       "GW                0.00         0.00        0.00        0.00         0.00\n",
       "KE                0.00         0.00       76.04        0.00         0.00\n",
       "UGgam             0.00         0.00      100.00        0.00         0.00"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "for p in phase2_ar1.pop_ids:\n",
    "    pop_df = pop_df.apply(lambda x: (x / (len(df_meta[df_meta.population == p]) * 2) * 100) if x.name == p else x, axis=1).round(decimals=2)\n",
    "pop_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "pop_df.to_csv('../tables/population_IR_frequency.csv', sep='\\t', index=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
